src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -55,42 +60,51 @@ public class SearchParameterOperations : ISearchParameterOperations
             _searchParameterSupportResolver = searchParameterSupportResolver;
             _dataStoreSearchParameterValidator = dataStoreSearchParameterValidator;
             _searchServiceFactory = searchServiceFactory;
+            _contextAccessor = contextAccessor;
             _logger = logger;
         }
 
         public async Task AddSearchParameterAsync(ITypedElement searchParam, CancellationToken cancellationToken)
         {
             try
             {
-                // verify the parameter is supported before continuing
                 var searchParameterWrapper = new SearchParameterWrapper(searchParam);
-                var searchParameterInfo = new SearchParameterInfo(searchParameterWrapper);
+                bool duplicateSearchParameter = IsSearchParameterDuplicate();
 
-                if (searchParameterInfo.Component?.Any() == true)
+                if (!duplicateSearchParameter)
                 {
-                    foreach (SearchParameterComponentInfo c in searchParameterInfo.Component)
+                    // verify the parameter is supported before continuing
+                    var searchParameterInfo = new SearchParameterInfo(searchParameterWrapper);
+
+                    if (searchParameterInfo.Component?.Any() == true)
                     {
-                        c.ResolvedSearchParameter = _searchParameterDefinitionManager.GetSearchParameter(c.DefinitionUrl.OriginalString);
+                        foreach (SearchParameterComponentInfo c in searchParameterInfo.Component)
+                        {
+                            c.ResolvedSearchParameter = _searchParameterDefinitionManager.GetSearchParameter(c.DefinitionUrl.OriginalString);
+                        }
                     }
-                }
 
-                (bool Supported, bool IsPartiallySupported) supportedResult = _searchParameterSupportResolver.IsSearchParameterSupported(searchParameterInfo);
+                    (bool Supported, bool IsPartiallySupported) supportedResult = _searchParameterSupportResolver.IsSearchParameterSupported(searchParameterInfo);
 
-                if (!supportedResult.Supported)
-                {
-                    throw new SearchParameterNotSupportedException(searchParameterInfo.Url);
-                }
+                    if (!supportedResult.Supported)
+                    {
+                        throw new SearchParameterNotSupportedException(searchParameterInfo.Url);
```

Comment:
Customer sees this message "The search parameter with definition URL '{0}' is not supported.".  I know this change is not related to this PR but in this scenario, error is thrown if definition has any issues. Do you think we can use better error message to indicate customer that something is wrong with the definition they have sent?